File: src-tauri/src/models/recipe_detail.rs
use serde::{Deserialize, Serialize};
use sqlx::prelude::FromRow;

#[derive(Debug, Deserialize, FromRow, Serialize, Clone)]
pub struct RecipeDetail {
    pub id: String,
    pub input_amount: f64,
    pub item_id: String,
    pub recipe_id: String,
}

#[derive(Debug, Deserialize, FromRow, Serialize, Clone)]
pub struct RecipeDetailDto {
    pub input_amount: f64,
    pub item_id: String,
    pub recipe_id: Option<String>,
}

#[derive(Debug, Deserialize, FromRow, Serialize, Clone)]
pub struct RecipeDetailForRecipe {
    pub id: String,
    pub input_amount: f64,
    pub item_id: String,
    pub recipe_id: String,
    pub item_picture: String,
}

File: src/models/recipe-detail.ts
interface RecipeDetail {
    id: string;
    input_amount: number;
    item_id: string;
    recipe_id: string;
}

interface RecipeDetailDto {
    input_amount: number;
    item_id: string;
    recipe_id?: string;
}

interface RecipeDetailForRecipe {
    id: string;
    input_amount: number;
    item_id: string;
    recipe_id: string;
    item_picture: string;
    item_picture_b64: string;
}